lib/auth/dbobjectimportrule/dbobjectimportrulev1/service.go

```diff
@@ -0,0 +1,194 @@
+)
+
+// Backend encapsulates required backend methods.
+type Backend interface {
+	services.DatabaseObjectImportRule
+}
+
+// DatabaseObjectImportRuleServiceConfig holds configuration options for
+// the DatabaseObjectImportRules gRPC service.
+type DatabaseObjectImportRuleServiceConfig struct {
+	Authorizer authz.Authorizer
+	Backend    Backend
+	Logger     logrus.FieldLogger
+}
+
+// NewDatabaseObjectImportRuleService returns a new instance of the DatabaseObjectImportRuleService.
+func NewDatabaseObjectImportRuleService(cfg DatabaseObjectImportRuleServiceConfig) (*DatabaseObjectImportRuleService, error) {
+	if cfg.Authorizer == nil {
+		return nil, trace.BadParameter("authorizer is required")
+	}
+	if cfg.Backend == nil {
+		return nil, trace.BadParameter("backend service is required")
+	}
+	if cfg.Logger == nil {
+		cfg.Logger = logrus.WithField(trace.Component, "DatabaseObjectImportRule.service")
+	}
+	return &DatabaseObjectImportRuleService{
+		logger:     cfg.Logger,
+		authorizer: cfg.Authorizer,
+		backend:    cfg.Backend,
+	}, nil
+}
+
+// DatabaseObjectImportRuleService implements the teleport.dbobjectimportrule.v1.DatabaseObjectImportRuleService RPC service.
+type DatabaseObjectImportRuleService struct {
+	pb.UnimplementedDatabaseObjectImportRuleServiceServer
+
+	backend    Backend
+	authorizer authz.Authorizer
+	logger     logrus.FieldLogger
+}
+
+// GetDatabaseObjectImportRule gets a DatabaseObjectImportRule by name. It will throw an error if the DatabaseObjectImportRule does not exist.
+func (bs *DatabaseObjectImportRuleService) GetDatabaseObjectImportRule(ctx context.Context, req *pb.GetDatabaseObjectImportRuleRequest) (*pb.DatabaseObjectImportRule, error) {
+	_, err := authz.AuthorizeWithVerbs(
+		ctx, bs.logger, bs.authorizer, false, types.KindDatabaseObjectImportRule, types.VerbRead,
+	)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	if req.Name == "" {
+		return nil, trace.BadParameter("name: must be non-empty")
+	}
+	out, err := bs.backend.GetDatabaseObjectImportRule(ctx, req.Name)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return out, nil
+}
+
+// ListDatabaseObjectImportRules lists all DatabaseObjectImportRules.
+func (bs *DatabaseObjectImportRuleService) ListDatabaseObjectImportRules(
+	ctx context.Context, req *pb.ListDatabaseObjectImportRulesRequest,
+) (*pb.ListDatabaseObjectImportRulesResponse, error) {
+	_, err := authz.AuthorizeWithVerbs(
+		ctx, bs.logger, bs.authorizer, false, types.KindDatabaseObjectImportRule, types.VerbList,
+	)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	out, next, err := bs.backend.ListDatabaseObjectImportRules(ctx, int(req.PageSize), req.PageToken)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return &pb.ListDatabaseObjectImportRulesResponse{
+		Rules:         out,
+		NextPageToken: next,
+	}, nil
+}
+
+// CreateDatabaseObjectImportRule creates a new DatabaseObjectImportRule. It will throw an error if the DatabaseObjectImportRule already
```

Comment:
```suggestion
// CreateDatabaseObjectImportRule creates a new DatabaseObjectImportRule. It will return an error if the DatabaseObjectImportRule already
```
Here, I think that the interface also would benefit from having this contract explicitly stated in the documentation.